src/enochecker/nosqldict.py

```diff
@@ -158,7 +172,11 @@ def __setitem__(self, key: Any, value: Any) -> None:
         :param value: value in the dictionary
         """
         self.cache[key] = value
+        self.hash_cache[key] = value_to_hash(value)
+
+        self._upsert(key, value)
```

Comment:
Do we still need the `_upsert` call in this case?  Maybe just always store on `persist`?
That way we could even drop the hashing.
Of course, it could break if anything happens in the meantime, so it may be risky.

Reply:
we don't "need" it, but in the (probably most common?) case where we only assign once to the dictionary we would execute this `_upsert` but not the second one in the `__del__`-function, so imo there are probably not a lot of performance drawbacks.

As you mentioned, in case something goes wrong between the `__setitem__` and `__del__`, we would lose the changes otherwise